design/media_types.go

```diff
@@ -30,17 +30,20 @@ var workItem = a.MediaType("application/vnd.workitem+json", func() {
 	a.Description("A work item hold field values according to a given field type")
 	a.Attribute("id", d.String, "unique id per installation")
 	a.Attribute("version", d.Integer, "Version for optimistic concurrency control")
+	a.Attribute("position", d.Number, "Position of workitem")
 	a.Attribute("type", d.String, "Name of the type of this work item")
 	a.Attribute("fields", a.HashOf(d.String, d.Any), "The field values, according to the field type")
 
 	a.Required("id")
+	a.Required("position")
```

Comment:
Position should not be exposed at this level. It's part of fields as far as the domain is concerned. workitem.Workitem should define it as a separate attribute.